### ISSUE TITLE
Failing test on MacOS, OpenMP library is not found

### ISSUE THREAD

**Issue by ArseniyKholod** [sloede/openfhe-julia#42]
See https://github.com/sloede/openfhe-julia/actions/runs/8816068748/job/24199359449?pr=41

**Comment by ArseniyKholod** [comment]
Hi @sloede!
I'm not really pro in MacOS and GitHub CI, may be you could give me a hint. The problem in this issue is that on MacOS runners OpenMP library is not installed, so it should be installed in ci.yml. 
I install it using `brew install libomp`.
But CI on MacOS passes only if I also rebuild both `openfhe-development` and `libcxxwrap-julia`, but if they would be retrieved from the cache, linker fails to find installed OpenMP
```
ld: warning: search path '/usr/local/opt/libomp/lib' not found
ld: library 'omp' not found 
```
I'm not sure, how it woks with caching, does packages update every time they are compiled? I also saw, that there are several locations, where OpenMP could be saved, depending on if MacPorts ore Home-brew were used. For Home-brew location of OpenMP also differs depending on architecture. Does caching mechanism aware of these differences?
All CI I runned were on M1, that uses Home-brew, but if `openfhe-development` is retrieved from cache, while building `openfhe-julia` linker search for OpenMP in the `/usr/local/opt/libomp/lib` location in case of MacPorts.
I should probably update some environmental variables, but I tried all I could guess, it doesn't work.

I'm dealing first time with MacOS, so if you have some ideas from experience, it would be very helpful!

**Comment by sloede** [comment]
`macos-latest` was changed from `macos-12` to `macos-14-arm64` in [this commit](https://github.com/actions/runner-images/pull/9601/files). Can you please check if the errors disappear if you change `macos-latest` to `macos-12`?

**Comment by ArseniyKholod** [comment]
> `macos-latest` was changed from `macos-12` to `macos-14-arm64` in [this commit](https://github.com/actions/runner-images/pull/9601/files). Can you please check if the errors disappear if you change `macos-latest` to `macos-12`?

That's great! The problem vanishes. You found out from the first shot! 🥇 :) Could we just replace `macos-latest` by `macos-12`?

**Comment by sloede** [comment]
> Could we just replace `macos-latest` by `macos-12`?

No. I don't think this is a viable strategy, since `macos-12` is to be deprecated soon(ish). We should get this up and running again with `macos-latest`. While I merged #41 such that you can get on with the other PRs, I am reopening this issue here as a reminder that we need to find a proper fix (and soon!).

**Comment by ArseniyKholod** [comment]
Thank you!
